lib/va_notify/notification_email/saved_claim.rb

```diff
@@ -0,0 +1,155 @@
+# frozen_string_literal: true
+
+require 'va_notify/notification_email'
+
+require 'pry-byebug'
+
+module VANotify
+  module NotificationEmail
+    # general SavedClaim email notification function
+    #
+    # an entry should be in Settings.vanotify.services - config/settings.yml
+    # @example
+    #     21p_527ez: &vanotify_services_pension
+    #       api_key: fake_secret
+    #       email:
+    #         confirmation:
+    #           template_id: form527ez_confirmation_email_template_id
+    #           flipper_id: false
+    #         error: null
+    #         received: null
+    #     pensions: *vanotify_services_pension
+    #
+    class SavedClaim
+      # constructor
+      #
+      # @param saved_claim [SavedClaim] the claim for which to send a notification
+      # @param service_name [String] alternative serivce name listed in Settings
+      #   default will be the formatted claim.form_id
+      #   @see #vanotify_service
+      def initialize(saved_claim, service_name: nil)
+        @claim = saved_claim
+        @vanotify_service = service_name
+      end
+
+      # deliver a notification for _claim_
+      # @see VANotify::EmailJob
+      # @see ClaimVANotification
+      #
+      # @param email_type [Symbol] one of VANotify::NotificationEmail::Type and defined in Settings
+      # @param at [String|DateTime] valid date string to schedule sending of notification
+      #   @see VANotify::EmailJob#perform_at
+      #
+      # @return [ClaimVANotification] db record of notification sent
+      def deliver(email_type, at: nil)
+        email_template_id = valid_attempt?(email_type)
+        return unless email_template_id
+
+        at ? enqueue_email(email_template_id, at) : send_email(email_template_id)
+
+        db_record = claim.insert_notification(email_template_id)
+        tags, context = monitoring(email_type)
+        VANotify::NotificationEmail.monitor_deliver_success(tags:, context:)
+
+        db_record
+      rescue => e
+        tags, context = monitoring(email_type)
+        VANotify::NotificationEmail.monitor_send_failure(e&.message, tags:, context:)
+      end
+
+      private
+
+      attr_reader :claim, :email_template_id
+
+      # return or default the service_name to be used
+      def vanotify_service
+        @vanotify_service ||= claim.form_id.downcase.gsub(/-/, '_')
```

Comment:
I suspect there will never be nils to cause a problem here so we don't need to:
`claim&.form_id&.downcase&.gsub(/-/, '_')` ?

Reply:
hmm, i would hope not, but there is a chance bad data could get passed in.  in theory it should be caught during testing (the vanotify.service would not be found and raise an error) but worth safeguarding